apps/api/src/app/partner-integrations/usecases/process-vercel-webhook/process-vercel-webhook.usecase.ts

```diff
@@ -24,79 +24,83 @@ export class ProcessVercelWebhook {
   ) {}
 
   async execute(command: ProcessVercelWebhookCommand) {
+    const eventType = command.body.type;
+    if (eventType !== 'deployment.succeeded') {
+      Logger.log({ eventType }, `Skipping processing Vercel webhook event: ${eventType}`);
+
+      return true;
+    }
+
+    const teamId = command.body.payload.team.id;
+    const projectId = command.body.payload.project.id;
+    const deploymentUrl = command.body.payload.deployment.url;
+    const vercelEnvironment = command.body.payload.target || 'preview';
+
     Logger.log(
       {
-        teamId: command.teamId,
-        projectId: command.projectId,
-        vercelEnvironment: command.vercelEnvironment,
-        deploymentUrl: command.deploymentUrl,
+        teamId,
+        projectId,
+        vercelEnvironment,
+        deploymentUrl,
       },
-      `Processing vercel webhook for ${command.vercelEnvironment}`
+      `Processing vercel webhook for ${vercelEnvironment}`
     );
 
     this.verifySignature(command.signatureHeader, command.body);
 
-    const url = command.deploymentUrl;
-
     const organizations = await this.organizationRepository.find(
       {
-        'partnerConfigurations.teamId': command.teamId,
-        'partnerConfigurations.projectIds': command.projectId,
+        'partnerConfigurations.teamId': teamId,
+        'partnerConfigurations.projectIds': projectId,
       },
       { 'partnerConfigurations.$': 1 }
     );
 
-    const organization = organizations[0];
-
-    if (!organization) {
-      Logger.error(
-        {
-          teamId: command.teamId,
-          projectId: command.projectId,
-        },
-        'Organization not found for vercel webhook integration'
-      );
+    if (!organizations || organizations.length === 0) {
+      Logger.error({ teamId, projectId }, 'Organization not found for vercel webhook integration');
 
       throw new ApiException('Organization not found');
     }
 
-    let environment: EnvironmentEntity | null;
-
-    if (command.vercelEnvironment === 'production') {
-      environment = await this.environmentRepository.findOne({
-        _organizationId: organization._id,
-        name: 'Production',
-      });
-    } else {
-      environment = await this.environmentRepository.findOne({
-        _organizationId: organization._id,
-        name: 'Development',
+    for (const organization of organizations) {
+      let environment: EnvironmentEntity | null;
+
+      if (vercelEnvironment === 'production') {
```

Comment:
Let's add a TODO here to handle about how to map Vercel Integration with other custom environments that are not dev or prod.